woocommerce/woocommerce
# Add card last 4 to order totals payment method row

### Changes proposed in this Pull Request:

Adds the last 4 digits of the payment card to the "Payment Method" field in the order details section of emails and customer-facing screens about their orders. Currently this only works for orders with successful payments--failed payments do not have the necessary data to display `last4`.

More context: p7bje6-6ob-p2#comment-9549

Also, this does not currently include the SVG card icon for the payment method the way that the printable receipts do. When I tried to include it, it kept getting stripped out of the email HTML. I suspect that data URIs are not very compatible with email, but I haven't dug any deeper into this at this point...

Fixes #52308

| Account Screen | Email |
|--------|--------|
| ![order-details-last4](https://github.com/user-attachments/assets/1691240b-52ad-4b65-9b61-5857b9e61c09) | ![payment-method-last4](https://github.com/user-attachments/assets/2d5df7fc-e93b-4f9d-b902-87c6a9770db6) |

### How to test the changes in this Pull Request:

1. Follow the first two steps of the instructions in #52173 so that you have an order created with a successful payment via WC Payments.
2. On the front end, go to the My Account screen, then Orders, and view the order you just made. The "Payment Method" row should say it was a credit card, and include the last 4 digits of the card.
3. Install a plugin like [WP Mail Logging](https://wordpress.org/plugins/wp-mail-logging/) to capture outgoing transactional emails from your test site.
4. Go to the Edit Order screen for the order you made. Under "Order Actions", choose "Send order details to customer".
5. Check the mail logger and find the email with the order details. It should also contain a "Payment Method" row that has the last 4 digits of the credit card.
6. You can try the same steps with a failed order, where you use a test card that will get declined. Notice that the last 4 digits are not included, because the necessary data about the failed transaction is not available for retrieving this info.
